src/ProducerSideService/include/ProducerSideService/BuildAndStartProducerSideServer.h

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_PRODUCER_SIDE_SERVICE_BUILD_AND_START_PRODUCER_SIDE_SERVER_H_
+#define ORBIT_PRODUCER_SIDE_SERVICE_BUILD_AND_START_PRODUCER_SIDE_SERVER_H_
+
+#include "ProducerSideService/ProducerSideServer.h"
+
+namespace orbit_producer_side_service {
+[[nodiscard]] std::unique_ptr<ProducerSideServer> BuildAndStartProducerSideServer();
```

Comment:
I think `BuildAndStartProducerSideServerWithUri` should also be made available here because for the cloud collector we probably wanna use a different unix domain socket to not clash with OrbitService.

And then I'm thinking it might be better to only make `BuildAndStartProducerSideServerWithUri` available here and leave `BuildAndStartProducerSideServer` in the service since it would be OrbitService-specific then.

@dpallotti @vickyliu-go4it WDYT?

Reply:
Uhh circling back on this idea. Is that even possible. I assume the socket is hard coded in UserSpace/ManualInstrumentation and VulkanLayer and can't be dynamically set? How do we wanna handle this?

- We could use the same socket and "embrace" the crash - so basically refuse to capture if the socket path is already occupied. (This might be even a good thing)
- We could make the clients trying to connect to multiple hard-coded sockets for availability.

Reply:
Thanks for pointing this out! I didn't realize we could crash ObritService in this case.

How about that let `BuildAndStartProducerSideServer` have a `string_view` argument `unix_domain_socket`.

Reply:
> I assume the socket is hard coded in UserSpace/ManualInstrumentation and VulkanLayer and can't be dynamically set? 

For vulkan layer and user space instrumentation producers, we use the default `unix_domain_socket_path`, i.e., "/tmp/orbit-producer-side-socket" while calling `CreateProducerSideChannel` to create the channel. So I think we might also able to make the socket path configurable. 

I'm fine with both directions, i.e., reuse the socket path or use multiple paths.

Reply:
I think we should wait for Davide's input before making a decision.

Reply:
As @vickyliu-go4it said, the socket is specified in the `ProducerSideChannel` module by `orbit_producer_side_channel::CreateProducerSideChannel()`. The external producers (Vulkan layer, user space instrumentation, manual instrumentation) call `LockFreeBufferCaptureEventProducer::BuildAndStart` and pass that.

I'm not sure we can fit the entire discussion here, so I'll just write down some thoughts.

I also don't really have a clear opinion. From a quick thought I think that having a single socket path is generally simpler but there are complications either way:

Keeping only one socket path and not allowing the cloud collector or `OrbitService` to start (or properly report an error) if another is already using the socket seems to be the "less surprising" solution. This would also be simpler from the point of view of the producers. 
But note the following: Currently "using" the same socket multiple times doesn't lead to any error or crash, and when that happens the behavior is unclear. For example, if you start two instances of `OrbitService`, both seem to succeed in creating the `ProducerSideService`, but only one will receive the connection (and data) from the producers. I'm not entirely sure what the logic is, although the instance started second seems to usually "win". So this would need to be investigated and handled better. Simply checking for the existence of the socket in the filesystem is probably not enough in case `OrbitService` crashed and left the socket behind.

Having multiple possible socket path brings some questions. I think the general one is: "What should happen when both `OrbitService` and the cloud collector are running?" For example: Should one have the priority over the other? In that case, what if one is started, the producer connect, and only then the one with higher priority is also started?
Also, handling the case where the producer is first connected to, say, `OrbitService`, then this exists and the cloud collector comes up would require some change in `orbit_capture_event_producer::CaptureEventProducer` as we always assume the same channel, but this might be trivial by simply recreating `producer_side_service_stub_` with a new channel on error.

Reply:
Meeting?

Reply:
Yes!

Reply:
I set something up.